osu.Game/Screens/Play/Player.cs

```diff
@@ -93,9 +93,9 @@ public abstract class Player : ScreenWithBeatmapBackground, ISamplePlaybackDisab
         [Resolved]
         private SpectatorClient spectatorClient { get; set; }
 
-        protected Ruleset GameplayRuleset { get; private set; }
+        public Ruleset GameplayRuleset { get; private set; }
 
-        protected GameplayBeatmap GameplayBeatmap { get; private set; }
+        public GameplayBeatmap GameplayBeatmap { get; private set; }
```

Comment:
Is it feasible that we just expose `GameplayBeatmap` via DI and get the ruleset via that, rather than direction through `Player`? Really don't like having central classes like `Player` as a requirement for getting to these.